### DISCUSSION TITLE
Beginner Questions

### DISCUSSION THREAD

**Discussion by PalusiOld** [unoplatform/uno#12827] [Q&A] [answered]
After I learned how to build apps with wpf i want to lean how to build apps with uno platform. I know have the following questions:

1. How can I change the application icon of my uno app?
2. How can I customize the splash menu for wasm?
3. How can I write System specific xaml and c#  code?
4. How can I write seperate Ui pages for different systems? For example if the mobile apps should have a different ui than the wasm or windows apps
5. Where can i find the finished build applications to deliver?

I already tried to customize the splash screen by editing the AppManifest.js file, but it does not make a difference.
I noticed that i can write anything in this file without getting an error and making a difference.

**I also tried to write system specific code**

this does not work:
```
using Microsoft.UI;

namespace MyUnoApp
{
    public sealed partial class MainPage : Page
    {
        public MainPage()
        {
            this.InitializeComponent();
        }

        private void Button_Click(object sender, RoutedEventArgs e)
        {
            string message = string.Empty;

#if NETFX_CORE
    message = "Hello from UWP World!";
#elif __ANDROID__
    message = "Hello from Android World!";
#elif __IOS__
    message = "Hello from iOS World!";
#elif __WASM__
    message = "Hello from Web Assembly World!";
#elif __MACOS__
    message = "Hello from MacOS World!";
#endif
            HelloTextBlock.Text = message;
        }
    }
}
```

and here only the windows part worked:
```
<Page x:Class="MyUnoApp.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:MyUnoApp"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
      xmlns:wasm="http://uno.ui/wasm"
      xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <StackPanel HorizontalAlignment="Center"
                VerticalAlignment="Center">
        <TextBlock
                   win:Text="by win"  x:Name="HelloTextBlock" />
        <TextBlock wasm:Text="by wasm"/>
        <Button Width="100" Height="100" Click="Button_Click"/>
        
    </StackPanel>
</Page>
```

**Would be nice if someone can help me getting started**

**Comment by jeromelaban** [comment, marked as answer]
> After I learned how to build apps with wpf i want to lean how to build apps with uno platform. I know have the following questions:
> 
> 1. How can I change the application icon of my uno app?

[This](https://platform.uno/docs/articles/external/uno.resizetizer/doc/using-uno-resizetizer.html#unoicon) should help.
> 2. How can I customize the splash menu for wasm?

[This](https://platform.uno/docs/articles/external/uno.resizetizer/doc/using-uno-resizetizer.html#unosplashscreen) should help.
> 3. How can I write System specific xaml and c#  code?

We've [documented this here](https://platform.uno/docs/articles/platform-specific-csharp.html), though there's a catch with `__WASM__` that is currently not properly documented. Wasm is a `net7.0` target, and cannot yet be discriminated at compile time, at least not until https://github.com/dotnet/designs/pull/289 will be available in .NET 8.

Until then, you'll need to use something like this:

```csharp
if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))
{
   // Do something WebAssembly specific
}
```

> 4. How can I write seperate Ui pages for different systems? For example if the mobile apps should have a different ui than the wasm or windows apps

You can do this using runtime checks, like the one specified above. You can also use [x:Load](https://platform.uno/docs/articles/features/windows-ui-xaml-xbind.html) to activate parts of your UI conditionally.

> 5. Where can i find the finished build applications to deliver?

It depends on the targets, but in general, `dotnet publish` will provide you with the output you need, regardless of the target platform. You can find [an example of a CI build here](https://github.com/unoplatform/Uno.QuickStart/blob/9242bdc9e0c467fdfaf135cfe07f3f3f1ded1178/.github/workflows/dotnet.yml#L34).

**Reply by PalusiOld** [reply]
Thank you, this helped me really a lot.

Now that I've changed the splash screen, I'm just wondering if I can shorten the loading time of the wasm application. For example, if I want to load a simple UI with a text, a button and no significant code behind, the splash screen is displayed for 20 seconds. Is this correct or is there a good way to speed this up?

**Reply by jeromelaban** [reply]
Are you starting the app with the debugger (Using F5 in visual studio) ? If so, this is likely the issue. If you build your app in release mode, and serve the output (the dist folder) with a small web server, like [dotnet-serve](https://github.com/natemcmaster/dotnet-serve), or in the cloud using [Azure static apps](https://platform.uno/docs/articles/guides/azure-static-webapps.html), the app will start faster.

Here are some other [performance tips](https://platform.uno/docs/articles/Uno-UI-Performance.html).
